### PR TITLE
Add AvroDecoderModule

### PR DESCRIPTION
**Pull request by elonazoulay**

### CODE REVIEW

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java [hunk 1]
```diff
@@ -103,12 +102,6 @@ public Builder setExtraTopicDescription(Map<SchemaTableName, KafkaTopicDescripti
             return this;
         }
 
-        public Builder setExtension(Module extension)
```
Comment (kokosing) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java:106]: Please undo, this is used externally.

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java [hunk 2]
```diff
@@ -161,18 +172,13 @@ private static DistributedQueryRunner createKafkaQueryRunner(
                     .putAll(tpchTopicDescriptions)
                     .putAll(testTopicDescriptions.build())
                     .build();
-            KafkaPlugin kafkaPlugin = new KafkaPlugin(combine(
-                    extensions,
-                    installModuleIf(
-                            KafkaConfig.class,
-                            kafkaConfig -> kafkaConfig.getTableDescriptionSupplier().equalsIgnoreCase(TEST),
-                            binder -> binder.bind(TableDescriptionSupplier.class)
-                                    .toInstance(new MapBasedTableDescriptionSupplier(topicDescriptions)))));
```
Comment (kokosing) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/KafkaQueryRunner.java:170]: How about:
```
 KafkaPlugin kafkaPlugin = new KafkaPlugin(combine(
                    extensions,
                    installModuleIf(
                            KafkaConfig.class,
                            kafkaConfig -> kafkaConfig.getTableDescriptionSupplier().equalsIgnoreCase(TEST),
                            binder -> binder.bind(TableDescriptionSupplier.class)
                                    .toInstance(new MapBasedTableDescriptionSupplier(topicDescriptions))),
                    new AvroModule()));
```

?

File: presto-record-decoder/src/main/java/io/prestosql/decoder/DecoderModule.java [hunk 3]
```diff
@@ -43,7 +41,6 @@ public void configure(Binder binder)
         decoderFactoriesByName.addBinding(CsvRowDecoder.NAME).to(CsvRowDecoderFactory.class).in(SINGLETON);
         decoderFactoriesByName.addBinding(JsonRowDecoder.NAME).to(JsonRowDecoderFactory.class).in(SINGLETON);
         decoderFactoriesByName.addBinding(RawRowDecoder.NAME).to(RawRowDecoderFactory.class).in(SINGLETON);
-        decoderFactoriesByName.addBinding(AvroRowDecoder.NAME).to(AvroRowDecoderFactory.class).in(SINGLETON);
```
Comment (kokosing) [inline presto-record-decoder/src/main/java/io/prestosql/decoder/DecoderModule.java:46]: I don't like it. I haven't noticed that earlier, I am sorry about it.

Previously `DecoderModule` was self contained. Now it has a dependency that you need to provide externally `AvroModule`. That could be non intuitive. Maybe let's make `DecoderModule` constructor to accept a module that is expected to provide Avro decoders, then default constructor just use `AvroDecoderModule`. WDYT? Then you don't need to change other connectors.

File: presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroBytesDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import static io.prestosql.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.util.Objects.requireNonNull;
+
+public class AvroBytesDeserializer<T>
```
Comment (kokosing) [inline presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroBytesDeserializer.java:27]: This class is not yet used in this PR.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTableDescriptionSupplierModule.java [hunk 5]
```diff
@@ -16,6 +16,7 @@
 import com.google.inject.Binder;
```
Comment (kokosing) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/schema/file/FileTableDescriptionSupplierModule.java:16]: Please change commit message to:
```
Extract AvroDecoderModule
```

Also describe why you are doing this, where you can say that this allows to use different implementation of avro decoders etc.

File: presto-record-decoder/src/main/java/io/prestosql/decoder/DecoderModule.java (hunk 3 above)
Comment (elonazoulay) [reply presto-record-decoder/src/main/java/io/prestosql/decoder/DecoderModule.java:46]: Sounds good!

File: presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroBytesDeserializer.java (hunk 4 above)
Comment (elonazoulay) [reply presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroBytesDeserializer.java:27]: Nice, I will remove it and add it back to the other pr.

**Review by kokosing — APPROVED** [review]

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**